jupyterlab_git/git.py

```diff
@@ -870,14 +872,33 @@ async def init(self, current_path):
         Execute git init command & return the result.
         """
         cmd = ["git", "init"]
+        cwd = os.path.join(self.root_dir, current_path)
         code, _, error = await execute(
-            cmd, cwd=os.path.join(self.root_dir, current_path)
+            cmd, cwd=cwd
         )
+        if code == 0:
+            code, _, error = await self._maybe_run_actions('post_init', cwd)
 
         if code != 0:
             return {"code": code, "command": " ".join(cmd), "message": error}
         return {"code": code}
 
+    async def _maybe_run_actions(self, name, cwd):
+        code = 0
+        stdout = None
+        stderr = None
+        if name in self.user_custom_actions:
+            actions_list = self.user_custom_actions[name]
+            for action in actions_list:
+                # We trust the actions as they were passed via a config and not the UI
+                code, stdout, stderr = await execute(
+                    shlex.split(action), cwd=cwd
+                )
```

Comment:
```suggestion
                try:
                    code, stdout, stderr = await execute(
                        shlex.split(action), cwd=cwd
                    )
                except CalledProcessError:
                    break
```

Should we also protect against subprocess failure or do we consider it corrupt the base git action?

Reply:
I caught all exceptions here and bailed.